### DISCUSSION TITLE
Try to connect Weasis with our PACS

### DISCUSSION THREAD

**Discussion by Makhaon** [nroduit/Weasis#388] [Q&A]
Hi!
We are developping a new feature in our PACS - Web DICOM. And faced issue with opening study.
Study list opened whell, but any studies does not opens.
Send you connections logs.
1. Study list request
`Method: GET Document: /studies Params: includefield=00080020,00080030,00080050,00080061,00080090,00081030,00100010,00100020,00100021,00100030,00100040,0020000D,00200010 limit=20 offset=0`

Response

`Replied 200 code. Content: [{"00201208":{"vr":"IS","Value":["1"]},"00201206":{"vr":"IS","Value":["1"]},"00200010":{"vr":"SH","Value":[""]},"0020000D":{"vr":"UI","Value":["1.3.55555.413.777722222.777720051124.7777092652"]},"00081030":{"vr":"LO","Value":[""]},"00080090":{"vr":"PN","Value":["PHYSICIAN"]},"00080061":{"vr":"CS","Value":["CR","OT","PR"]},"00080056":{"vr":"CS","Value":["ONLINE"]},"00080050":{"vr":"SH","Value":["20051124092652"]},"00080030":{"vr":"TM","Value":["092652"]},"00080020":{"vr":"DA","Value":["20051124"]},"00080005":{"vr":"CS","Value":["ISO_IR 192"]},"00100040":{"vr":"CS","Value":[""]},"00100030":{"vr":"DA","Value":["20051124"]},"00100020":{"vr":"LO","Value":["ID_0000001"]},"00100010":{"vr":"PN","Value":["Belova B.B."]}}]`

2. Series list request
`Method: GET Document: /studies/1.3.55555.413.777722222.777720051124.7777092652/series Params: includefield=0008103E,00080060,0020000E,00200011,00081190`

Response

`Replied 200 code. Content: [{"00201209":{"vr":"IS","Value":["1"]},"00200011":{"vr":"IS","Value":["1"]},"0020000E":{"vr":"UI","Value":["1.3.55555.413.8888.20051124.092903"]},"0008103E":{"vr":"LO","Value":["Series 20051108125654"]},"00080060":{"vr":"CS","Value":["CR"]},"00080005":{"vr":"CS","Value":["ISO_IR 192"]}}]`

It looks like everything is working well.

Trying to open study with 1 image

1. Series list request
`Method: GET Document: /studies/1.3.55555.413.777722222.777720051124.7777092652/series Params: includefield=0008103E,00080060,0020000E,00200011,00081190`

Response

`Replied 200 code. Content: [{"00201209":{"vr":"IS","Value":["1"]},"00200011":{"vr":"IS","Value":["1"]},"0020000E":{"vr":"UI","Value":["1.3.55555.413.8888.20051124.092903"]},"0008103E":{"vr":"LO","Value":["Series 20051108125654"]},"00080060":{"vr":"CS","Value":["CR"]},"00080005":{"vr":"CS","Value":["ISO_IR 192"]}}]`

2. Image list request
`Method: GET Document: /studies/1.3.55555.413.777722222.777720051124.7777092652/series/1.3.55555.413.8888.20051124.092903/instances Params: includefield=00080018,00200013,00081190`

Response

`Replied 200 code. Content: [{"00200013":{"vr":"IS","Value":["1"]},"00081190":{"vr":"UR","Value":["https://127.0.0.1:82:5001/studies/1.3.55555.413.777722222.777720051124.7777092652/series/1.3.55555.413.8888.20051124.092903/instances/1.3.55555.413.8888.20051124.092903.1"]},"00080056":{"vr":"CS","Value":["ONLINE"]},"00080018":{"vr":"UI","Value":["1.3.55555.413.8888.20051124.092903.1"]}}]`

3. Thumbnail request
`Method: GET Document: /studies/1.3.55555.413.777722222.777720051124.7777092652/series/1.3.55555.413.8888.20051124.092903/thumbnail Params: viewport=256,256`

Send empty response, 200 code, send Jpeg as binary file with ContentType 'image/jpeg'.

And nothing more happened. No erros, no opens, no nothing.

Try to open thumbnail in the browser, opened successfully as Jpeg, size 256x256.

Response headers from our storage:

HTTP/1.1 200 OK
Connection: keep-alive
Server: SynHttpSrv/1.0
Content-Type: image/jpeg
Content-Length: 5312
Date: Wed, 12 Apr 2023 12:35:23 GMT

Try to switch log level to ERROR, but could not find where it stored.

Maybe you know what can be fixed here?

Also I try to open studies from Orthank storage, and Weasis works well:
https://demo.orthanc-server.com/dicom-web

I'm using v4.0.3 on Windows 10

**Comment by nroduit** [comment]
Unfortunately, I will only investigate if there are indications of a problem in the DICOMWeb implementation of Weasis.

Currently, there are no open bugs on this part and it works with many server implementations. If you think there is a problem please specify in which part of the protocol.

Many DICOMWeb server implementations do not have the Thumbnail part.

**Comment by Makhaon** [comment]
Thank you for response. I'm not saying the problem is on your side. Maybe our PACS does something wrong. But how is possible to know what is an issue?

**Comment by Makhaon** [comment]
I have installed latest Orthanc PACS. Setup connection and have same issue: patient list shows normally but Importing does not works. Please help me found reason of issue.

**Comment by nroduit** [comment]
As it is already difficult for me to find the time to answer requests for the projects I contribute to, I cannot assist in configuring or debugging other applications.

**Comment by Makhaon** [comment]
Ok, thank you for response anyway. I just want to ask question. Is it any way to debug issues in your or connected software? Some kind of logs or any other solution?

**Comment by nroduit** [comment]
You can refer to the documentation [to activate the log files](https://nroduit.github.io/en/faq/index.html#how-to-enable-weasis-logging) and the developer documentation [to debug the application](https://nroduit.github.io/en/getting-started/guidelines/index.html).
